### PR TITLE
test: add unit test coverage for Python ECDSA implementation

### PR DESCRIPTION
**Pull request by theStack**
This PR adds missing unit test coverage for the Python ECDSA implementation, which should be useful for detecting potential problems early whenever changes in the test framework's Python implementation of secp256k1 are made (e.g. #26222). Note that right now we don't call `ECPubKey.verify_ecdsa` anywhere in our tests, so we wouldn't notice if it is broken at some point.

To keep it simple, the already existing unit test for Schnorr signatures is extended to also check ECDSA signatures. For that purpose, the dictionary storing private-key/public-key entries use their legacy types `ECKey/ECPubKey` instead of bare byte-arrays, and for Schnorr signing/verification the necessary conversions (ECKey -> bare private key, ECPubKey -> x-only pubkey) is done later when needed. To avoid code duplication, a helper function `random_bitflip` for damaging signatures is introduced.

The unit test can be run by either calling it for this single module:
`$ python3 -m unittest ./test/functional/test_framework/key.py`
or simply running `$ ./test/functional/test_runner.py` which calls all test framework module's unit tests at the start (see TEST_FRAMEWORK_MODULES list).

### CODE REVIEW

File: test/functional/test_framework/key.py [hunk 1]
```diff
@@ -513,24 +513,33 @@ def sign_schnorr(key, msg, aux=None, flip_p=False, flip_r=False):
...
         keys = {}
-        for privkey in byte_arrays:  # build array of key/pubkey pairs
-            pubkey, _ = compute_xonly_pubkey(privkey)
-            if pubkey is not None:
-                keys[privkey] = pubkey
+        for privkey_bytes in byte_arrays:  # build array of key/pubkey pairs
+            privkey = ECKey()
+            privkey.set(privkey_bytes, compressed=True)
+            if privkey.is_valid:
+                keys[privkey] = privkey.get_pubkey()
```
Comment (pinheadmz) [inline test/functional/test_framework/key.py:529]: IIUC we don't assert that the keys we expect to be invalid are marked correctly. Seems like we mine as well ensure that `generate_privkey()` is always valid but certain edge cases are not (`0`, `SECP256K1_ORDER`, and `2**256 - 1`)

**Review by pinheadmz — COMMENTED** [review]
concept ACK

I also think it would be cool to see a set of ECDSA test vectors here. We already have a set of BIP340 vectors tested both in python and in the `key_tests.cpp` unit test, assuring that both implementations handle Schnorr (at least those cases) identically.

**Comment by theStack** [comment]
Rebased on master.

**Review by sipa — COMMENTED** [review]
utACK 96b3f2dbe4395ca55cfdd58c8f9f9bd7ca163983

**Review by stratospher — COMMENTED** [review]
tested ACK 96b3f2d.

**Comment by achow101** [comment]
ACK 96b3f2dbe4395ca55cfdd58c8f9f9bd7ca163983

**Pull request merged**